api_test.go

```diff
@@ -0,0 +1,70 @@
+package eos
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	mockserver "github.com/eoscanada/eos-go/testdata/mock_server"
+	"github.com/stretchr/testify/assert"
+)
+
+var api *API
+
+func TestGetAccount(t *testing.T) {
+	name := AccountName("teamgreymass")
+	acc, err := api.GetAccount(context.Background(), name)
+
+	assert.NoError(t, err)
+
+	actualJSON, err := json.Marshal(acc)
+	if err != nil {
+		panic(err)
+	}
+
+	expectedJSON := mockserver.OpenFile("chain_get_account.json")
+
+	assert.JSONEq(t, expectedJSON, string(actualJSON))
+}
+
+func TestAPIGetInfo(t *testing.T) {
+	info, err := api.GetInfo(context.Background())
+	if err != nil {
+		panic(fmt.Errorf("get info: %w", err))
+	}
+
+	assert.NoError(t, err)
+
+	actualJSON, err := json.Marshal(info)
+	if err != nil {
+		panic(err)
+	}
+
+	expectedJSON := mockserver.OpenFile("chain_get_info.json")
+
+	assert.JSONEq(t, expectedJSON, string(actualJSON))
+}
+
+func TestMain(m *testing.M) {
+	setUp()
+	code := m.Run()
+	tearDown()
+
+	os.Exit(code)
+}
```

Comment:
I've been doing Go for the last 5 years, first time I saw that (or at least that I remember) amazing!